odoo/addons/base/models/res_users.py

```diff
@@ -196,9 +196,11 @@ class Groups(models.Model):
     color = fields.Integer(string='Color Index')
     full_name = fields.Char(compute='_compute_full_name', string='Group Name', search='_search_full_name')
     share = fields.Boolean(string='Share Group', help="Group created to set access rights for sharing data with some users.")
+    apikey_duration = fields.Float(string='Duration (days)', required=True, default=1)
```

Comment:
![image](https://github.com/user-attachments/assets/633b233b-5a1c-47dd-9e05-274e74dcc9fe)
The label of the field is completely unclear currently.

```suggestion
    apikey_duration = fields.Float(string='API Keys maximum duration days', required=True, default=1)
```
I would also rename the technical name

I am also not sure we should display this with such importance in the form. It should be more hidden. Like in a tab or something.

```suggestion
    api_key_duration = fields.Float(string='API Keys maximum duration days', required=True, default=1)
```
And finally I am not sure I would put a default value. I would keep it empty.

```suggestion
    api_key_duration = fields.Float(string='API Keys maximum duration days', required=True)
```

And then to compute the maximum allowed, I would take the maximum of all groups where the value is set.